### PR TITLE
[SandboxIR] Update visibility of IR constructors.

### PR DESCRIPTION
**Pull request by vporpo**
All SandboxIR objects are owned by sandboxir::Context and should be created using sandboxir::Context's builder functions.
This patch protects the constructors to prohibit their use by the user.

### CODE REVIEW

File: llvm/unittests/SandboxIR/SandboxIRTest.cpp [hunk 1]
```diff
@@ -58,51 +46,66 @@ define void @foo(i32 %v1) {
...
 #ifndef NDEBUG
-  // The dump() functions should be very forgiving and should not crash even if
-  // sandboxir has not been built properly.
-  F.dump();
-  Arg0.dump();
-  Const0.dump();
-  OpaqueI.dump();
+  std::string Buff;
+  raw_string_ostream BS(Buff);
+  F->dump(BS);
```
Comment (aeubanks) [inline llvm/unittests/SandboxIR/SandboxIRTest.cpp:103]: should check the contents of these dumps, but can do that in separate PR

**Review by aeubanks — APPROVED** [review]

File: llvm/unittests/SandboxIR/SandboxIRTest.cpp (hunk 1 above)
Comment (vporpo) [reply llvm/unittests/SandboxIR/SandboxIRTest.cpp:103]: Yeah, I think I should add a test that checks all dumps.
